src/dune_pkg/opam_solver.ml

```diff
@@ -1816,21 +1816,24 @@ let reject_unreachable_packages =
             [ "name", Package_name.to_dyn name ]
         | Some (pkg : Lock_dir.Pkg.t), None -> Some (List.map pkg.depends ~f:snd)
         | None, Some (pkg : Local_package.For_solver.t) ->
-          let formula = Dependency_formula.to_filtered_formula pkg.dependencies in
-          (* Use `dev` because at this point we don't have any version *)
-          let opam_package =
-            OpamPackage.of_string (sprintf "%s.dev" (Package_name.to_string pkg.name))
-          in
-          let env = add_self_to_filter_env opam_package (Solver_env.to_env solver_env) in
-          let resolved =
-            Resolve_opam_formula.filtered_formula_to_package_names
-              env
-              ~with_test:true
-              (Package_name.Map.set pkgs_by_version Dune_dep.name dune_version)
-              formula
-          in
           let deps =
-            match resolved with
+            match
+              let env =
+                let opam_package =
+                  OpamPackage.create
+                    (Package_name.to_opam_package_name pkg.name)
+                    (* Use [dev] because at this point we don't have any version *)
+                    (Package_version.to_opam_package_version Package_version.dev)
+                in
+                Solver_env.to_env solver_env |> add_self_to_filter_env opam_package
+              in
+              Dependency_formula.to_filtered_formula pkg.dependencies
+              |> Resolve_opam_formula.filtered_formula_to_package_names
+                   ~env
+                   ~with_test:true
+                   ~packages:
+                     (Package_name.Map.set pkgs_by_version Dune_dep.name dune_version)
```

Comment:
Wouldn't it be more readable if env was computed outside the `match ... with`?

Reply:
Why do you think it would be more readable? This change reduces the scope of the identifiers. All else being equal, this make the code more readable.

Reply:
You are right, maybe it is because I'm more used to matching on variables instead of expression.